ludwig/features/image_feature.py

```diff
@@ -222,11 +226,23 @@ def _finalize_preprocessing_parameters(
                 )
         else:
             # User hasn't specified height and width.
-            # So we assume that all images have the same width and height.
-            # Thus the width and height of the first one are the same
-            # as all the other ones
-            height = first_img_height
-            width = first_img_width
+            # So infer height and width from a sample.
+            if INFER_IMAGE_DIMENSIONS in preprocessing_parameters and preprocessing_parameters[INFER_IMAGE_DIMENSIONS]:
+                should_resize = True
+                sample_size = min(len(input_feature_col), ImageFeatureMixin.IMAGE_SAMPLE_SIZE)
+                sample_images = [ImageFeatureMixin.read_image(
+                    ImageFeatureMixin.get_image_from_path(src_path, img)) for img in input_feature_col[:sample_size]]
+
+                height_avg = min(
```

Comment:
These operations should be max not min, as we want to make sure the inferred image size is not above a certain amount.

Also, having a default `ImageFeatureMixin.MAX_IMAGE_INFER_DIM = 256` is ok, but maybe that could be a preprocessing parameter too.